File: androidchallenge1/src/main/java/class28/androidchallenge01/ui/plates/ActRestaurantPlates.java
package class28.androidchallenge01.ui.plates;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import class28.androidchallenge01.R;
import class28.androidchallenge01.model.Restaurant;

import static class28.androidchallenge01.ui.restaurants.FragRestaurants.CHOSEN_RESTAURANT;


public class ActRestaurantPlates extends AppCompatActivity {

    public static final String CHOSEN_PLATE = "CHOSEN_PLATE";
    ImageButton restaurantDetailsBackButton;
    ImageView restaurantDetailsImage;
    TextView restaurantName;
    RecyclerView rvRestaurantPlates;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_restaurant_details);
        bindViews();
        restaurantDetailsBackButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });

        Restaurant restaurant = getIntent().getParcelableExtra(CHOSEN_RESTAURANT);

        restaurantDetailsImage.setImageResource(restaurant.getRestaurantImage());
        restaurantName.setText(restaurant.getName());
        rvRestaurantPlates.setLayoutManager(new GridLayoutManager(this, 2));
        rvRestaurantPlates.setAdapter(new AdapterRestaurantPlates(restaurant.getPlateList()));
    }

    private void bindViews() {
        restaurantDetailsBackButton = findViewById(R.id.restaurant_details_back_button);
        restaurantDetailsImage = findViewById(R.id.restaurant_details_image);
        restaurantName = findViewById(R.id.restaurant_details_name_title);
        rvRestaurantPlates = findViewById(R.id.rvRestaurantPlates);
    }
}

File: androidchallenge1kotlin/src/main/java/class28/androidchallenge01kotlin/ui/plates/ActRestaurantPlates.kt
package class28.androidchallenge01kotlin.ui.plates

import android.os.Bundle
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import class28.androidchallenge01kotlin.R
import class28.androidchallenge01kotlin.model.Restaurant
import class28.androidchallenge01kotlin.ui.restaurants.FragRestaurants.Companion.CHOSEN_RESTAURANT
import kotlinx.android.synthetic.main.activity_restaurant_details.*

class ActRestaurantPlates : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_restaurant_details)
        findViewById<ImageButton>(R.id.restaurant_details_back_button).setOnClickListener { onBackPressed() }

        val restaurant = intent.getParcelableExtra(CHOSEN_RESTAURANT) as Restaurant

        findViewById<ImageView>(R.id.restaurant_details_image).setImageResource(restaurant.restaurantImageRes)
        findViewById<TextView>(R.id.restaurant_details_name_title).text = restaurant.name

        rvRestaurantPlates.layoutManager = GridLayoutManager(this, 2)
        rvRestaurantPlates.adapter = AdapterRestaurantPlates(restaurant.plateList)
    }

    companion object {
        const val CHOSEN_PLATE = "CHOSEN_PLATE"
    }
}